app.rb

```diff
@@ -228,10 +228,20 @@ def events
 }.freeze
 
 
-use Rack::Auth::Basic do |username, password|
-  username == ENV.fetch('HTTP_BASIC_USER') and password == ENV.fetch('HTTP_BASIC_PASSWORD')
+helpers do
+  def protected!
+    return if authorized?
+    headers['WWW-Authenticate'] = 'Basic realm="Restricted Area"'
```

Comment:
Can we use the same [`unauthorized` helper that the `Rack::Auth::Basic` code calls](https://github.com/rack/rack/blob/master/lib/rack/auth/basic.rb#L21) to handle this response?

Reply:
I don't know. It looks like a private method.